fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java

```diff
@@ -873,8 +875,14 @@ && getTableProperty().getDynamicPartitionProperty().getBuckets()
                     long newTabletId = env.getNextId();
                     Tablet newTablet = EnvFactory.getInstance().createTablet(newTabletId);
                     idx.addTablet(newTablet, null /* tablet meta */, true /* is restore */);
-
                     // replicas
+                    if (Config.isCloudMode()) {
+                        long newReplicaId = Env.getCurrentEnv().getNextId();
+                        Replica replica = new CloudReplica(newReplicaId, null, ReplicaState.NORMAL,
+                                visibleVersion, schemaHash, db.getId(), id, partition.getId(), idx.getId(), i);
+                        newTablet.addReplica(replica, true /* is restore */);
+                        continue;
+                    }
                     try {
```

Comment:
May be we can add two functions, createCloudReplicas and createReplicas to make code more clear.

Reply:
good idea

Reply:
Well, keep origin code in the first version.